apis/cloudantv1/v1alpha1/cloudantdatabase_types.go

```diff
@@ -25,26 +25,27 @@ import (
 // CloudantDatabaseParameters are the configurable fields of a CloudantDatabase.
 type CloudantDatabaseParameters struct {
 
+	// should this json tag be 'json:"db"' ??
 	// The name of the database
 	// +immutable
 	Db string `validate:"required,ne="`
 
 	// Query parameter to specify whether to enable database partitions when creating a database.
-	// +immutable not doing update so these are immutable right ??
+	// +immutable
 	// +optional
 	Partitioned *bool `json:"partitioned,omitempty"`
 
 	// The number of shards in the database. Each shard is a partition of the hash value range. Default is 8, unless
 	// overridden in the `cluster config`.
-	// +immutable not doing update so these are immutable right ??
+	// +immutable
 	// +optional
 	Q *int64 `json:"q,omitempty"`
 }
 
 // CloudantDatabaseObservation are the observable fields of a CloudantDatabase.
+// are the json tags correct or should they all be omitempty for observation ??
 type CloudantDatabaseObservation struct {
 
-	// should the json be omitempty here and all other validate:"required" in observation ??
 	// Schema for database cluster information.
```

Comment:
I don't think we need validate:"required" in the Crossplane provider code. Just use "omitempty"